regression-tests/pure2-template-specialization.cpp2

```diff
@@ -0,0 +1,18 @@
+t: @struct <T: type> type = {
+  a: i32 = 1;
+}
+t: @struct <T: type> type < T> type requires std::is_void_v < T> = {
+  b: i32 = 2;
+}
+t: @struct type < i64> type = {
+  c: i32 = 3;
+}
+v: <T> i32 = 1; // clang-format off
+v: <> type<void> i32 = 2; // clang-format on
```

Comment:
I got linker errors for `c++ (GCC) 13.2.1 20230728 (Red Hat 13.2.1-1)`. The symbol `v<int>` did not exist.

With a specialization for i32 it worked for me.

```suggestion
v: <T> i32 = 0; // clang-format off
v: <> type<i32> i32 = 1;
v: <> type<void> i32 = 2; // clang-format on
```

Reply:
That sounds like a compiler bug.
It should instantiate the primary template.

Reply:
I got the same result locally,
and it reminded me of <https://github.com/hsutter/cppfront/pull/480#issuecomment-1569450825>.